libs/datagrid/src/lib/cells/order-type-cell.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { MouseEvent } from 'react';
+import { useMemo } from 'react';
+import { useCallback } from 'react';
+import get from 'lodash/get';
+import { t } from '@vegaprotocol/i18n';
+import * as Schema from '@vegaprotocol/types';
+
+interface OrderTypeCellProps {
+  value?: Schema.OrderType;
+  data?: {
+    id?: string;
+    marketId?: string;
+    market?: { id: string };
+    peggedOrder?: any;
+    liquidityProvision?: any;
+  };
+  idPath?: string;
+  onClick?: (marketId: string, metaKey?: boolean) => void;
+}
+
+export const OrderTypeCell = ({
+  value,
+  data: order,
+  idPath,
+  onClick,
+}: OrderTypeCellProps) => {
+  const id = order ? get(order, idPath ?? 'id', 'all') : '';
```

Comment:
suggestion (non blocking): I think this would be cleaner and simpler if the onClick function just passed the order object. Then the app can handle it however it pleases and you don't need to make sure the id path matches up

Reply:
I'd even say we could go further and pass the actual browser event too

Reply:
yes, cause this one is not reused. and it is very order type specific. 
for now only for liquidity provision orders we're rendering a link

Reply:
I updated it, it is very similar with market-name-cell but not as re-used so we don't need that id path.